examples/reddit_pagerank.cpp

```diff
@@ -0,0 +1,267 @@
+
+  map_type norm_map(world);
+  auto norm_map_ptr = norm_map.get_ygm_ptr();
+
+  auto compute_norm_lambda = [&pr_old_ptr, &pr_new_ptr, &norm_map_ptr](auto old_pr_pair) {
+
+    auto vtx_id = old_pr_pair.first;
+    auto pr_val = old_pr_pair.second; // old PR value.
+
+    auto visit_new_pr = [](auto &new_pr_pair, const auto old_val, auto &norm_map_ptr) {
+
+      auto vtx_id = new_pr_pair.first;
+      auto new_val = new_pr_pair.second;
+
+      auto diff = (new_val - old_val);
+      diff = diff * diff; 
+
+      norm_map_ptr->async_insert_if_missing(vtx_id, diff); 
+
+      #ifdef dbg
+      auto accumulate_lambda = [](auto &row_id_val, const auto &update_val) {
+        auto row_id = row_id_val.first;
+        auto value =  row_id_val.second;
+        auto append_val = value + update_val;
+        row_id_val.second = row_id_val.second + update_val;
+      };
+      norm_map_ptr->async_insert_if_missing_else_visit(std::string("dist"), diff, accumulate_lambda);
+      #endif
+
+    };
+
+    pr_new_ptr->async_visit(vtx_id, visit_new_pr, pr_val, norm_map_ptr);
+  };
+
+  pr_old_ptr->for_all(compute_norm_lambda);
+  world.barrier();
+
+  double norm_acc{0.};
+  auto print_res_lambda = [&norm_acc](auto res_kv_pair) {
+    norm_acc += res_kv_pair.second; 
+  };
+  norm_map.for_all(print_res_lambda);
+  double global_norm_acc = sqrt(world.all_reduce_sum(norm_acc));
+  std::cout << norm_acc << " " << global_norm_acc << std::endl;
+
+  return global_norm_acc;
+}
+
+int main(int argc, char **argv) {
+
+  ygm::comm world(&argc, &argv);
+
+  using map_type      = ygm::container::map<boost::json::string, double>;
+  using maptrix_type  = ygm::container::experimental::maptrix<boost::json::string, double>;
+  namespace ns_spmv   = ygm::container::experimental::detail::algorithms;
+
+  map_type pr(world);
+  map_type deg(world);
+  maptrix_type A(world);
+  
+  auto pr_ptr  = pr.get_ygm_ptr();
+  auto deg_ptr = deg.get_ygm_ptr();
+  auto A_ptr   = A.get_ygm_ptr(); 
+
+  if(argc == 1) {
+    std::cout << "Expected parameter arguments, exiting.." << std::endl;
+    exit(0);
+  }
+
+  auto A_acc_lambda = [](auto &row, auto &col, auto &value, const auto &update_val) {
+    value = value + update_val;
+  };
+
+  auto deg_acc_lambda = [](auto &rv_pair, const auto &update_val) {
+    rv_pair.second = rv_pair.second + update_val;
+  };
+
+  ygm::timer read_graph_timer{};
+
+  std::vector<std::string> fnames;
+  for (int i = 1; i < argc; ++i) {
+    fnames.push_back(argv[i]);
+    std::cout << argv[i];
+  }
+
+  // Building the author-author graph.
+  ygm::io::ndjson_parser json_parser(world, fnames);
+  json_parser.for_all([&A, &deg, &deg_acc_lambda, &A_acc_lambda](auto &json_line) {
+    if (json_line["LL_author"] != "AutoModerator" &&
+       json_line["LL_parent_author"] != "AutoModerator") {
+
+      /* Reading it in backward.. */
+      auto dst = json_line["LL_author"].as_string(); 
+      auto src = json_line["LL_parent_author"].as_string();
```

Comment:
This creates a copy of src and dst in each line. Declare these as `const auto &`